ui/components/multichain/network-list-menu/network-list-menu.js

```diff
@@ -62,7 +63,19 @@ export const NetworkListMenu = ({ onClose }) => {
 
   const completedOnboarding = useSelector(getCompletedOnboarding);
 
+  const [lineaMainnetReleased, setLineaMainnetReleased] = useState(false);
```

Comment:
This will result in a network request every time the network menu is rendered. We could instead call `shouldShowLineaMainnet` in the constructor of the preferences controller, and store the result in preferences controller state, so that once mainnet is released, further requests are not needed.

Reply:
Great point. Do we need the `eth_chainId` call for this? It would be ideal to avoid network call side-effects if possible, and just use the timer.

Reply:
We do need a network call, because they don't want to show the network before infura's Linea mainnet is actually live. So the logic is supposed to be: don't show the network before this date AND don't show it before infura's mainnet is live. The date based condition ensures that the network won't be shown before the planned launch date, the network call check allows the launch date to be moved.